src/NServiceBus.Core/Recoverability/Recoverability.cs

```diff
@@ -163,11 +170,57 @@ static void RaiseLegacyNotifications(FeatureConfigurationContext context)
             });
         }
 
+        void SetupLegacyRetriesSatellite(FeatureConfigurationContext context)
+        {
+            var retriesQueueLogicalAddress = context.Settings.LogicalAddress().CreateQualifiedAddress("Retries");
+            var retriesQueueTransportAddress = context.Settings.GetTransportAddress(retriesQueueLogicalAddress);
+
+            var mainQueueLogicalAddress = context.Settings.LogicalAddress();
+            var mainQueueTransportAddress = context.Settings.GetTransportAddress(mainQueueLogicalAddress);
+
+            var requiredTransactionMode = context.Settings.GetRequiredTransactionModeForReceives();
+            var maxNumberOfImmediateRetries = 4;
+
+            context.AddSatelliteReceiver("Legacy Retries Processor", retriesQueueTransportAddress, requiredTransactionMode, new PushRuntimeSettings(maxConcurrency: 1),
+                (config, errorContext) =>
+                {
+                    if (errorContext.ImmediateProcessingFailures <= maxNumberOfImmediateRetries)
```

Comment:
Nit pick: we could simplify this to just move it to error (non blocking for merge)